johnzon-jsonb/src/test/java/org/apache/johnzon/jsonb/AdapterTest.java

```diff
@@ -44,13 +47,21 @@ public void adapt() throws Exception {
             foo.bar.value = 1;
             foo.dummy = new Dummy();
             foo.dummy.value = 2L;
+            foo.baz = new Baz();
+            foo.baz.value = "3";
 
             final String toString = jsonb.toJson(foo);
-            assertEquals("{\"bar\":\"1\",\"dummy\":\"2\"}", toString);
+            assertThat(toString, startsWith("{"));
+            assertThat(toString, containsString("\"bar\":\"1\""));
+            assertThat(toString, containsString("\"dummy\":\"2\""));
+            assertThat(toString, containsString("\"baz\":\"3\""));
+            assertThat(toString, endsWith("}"));
+            assertEquals("{\"bar\":\"1\",\"dummy\":\"2\",\"baz\":\"3\"}".length(), toString.length());
 
             final Foo read = jsonb.fromJson(toString, Foo.class);
             assertEquals(foo.bar.value, read.bar.value);
             assertEquals(foo.dummy.value, read.dummy.value);
+            assertEquals(foo.baz.value, read.baz.value);
```

Comment:
I would also add a test where baz is the root object too, we had issues with that on other metadata by the past.

Reply:
I'll do